### DISCUSSION TITLE
About how a node can contain multiple scripts.

### DISCUSSION THREAD

**Discussion by teamjistory** [godotengine/godot-proposals#5609] [Engine Core]
About how a node can contain multiple scripts.

**Comment by Calinou** [comment]
There used to be a way to have a single node have multiple scripts attached to it (such as https://github.com/godotengine/godot/pull/8502, but it was removed eventually. You can use multiple child nodes or inheritance instead.

**Reply by teamjistory** [reply]
Yes, I just read and understood that part.
The advantage of one script (one component or one module or one file) for one node (component or object) is that it is easy for programmers to understand and maintain, apart from the system perspective.
However, I think a more granular script (components, files) and a separate code collection structure (multi-script) are good for using a good architecture and structure. Also, I think that the larger the project, the easier it will be to understand and maintain it.

Therefore, I would like you to review it again at this point.

**Reply by Zireael07** [reply]
The downside of multiscirpt is that the engine core becomes waaay more difficult to maintain, that's why multiscript was removed.

I recall a related proposal, something something mixins?

**Reply by Calinou** [reply]
> I recall a related proposal, something something mixins?

Indeed, see https://github.com/godotengine/godot-proposals/issues/758. However, I believe traits are the most likely solution to be implemented in the future (not for 4.0).

**Comment by teamjistory** [comment]
The advantage of one script (one component or one module or one file) for one node (component or object) is that it is easy for programmers to understand and maintain, apart from the system perspective.
However, I think a more granular script (components, files) and a separate code collection structure (multi-script) are good for using a good architecture and structure. Also, I think that the larger the project, the easier it will be to understand and maintain it.
Therefore, I would like you to review it again at this point.

**Reply by Zireael07** [reply]
you duplicated this comment - net hiccup?
